editor/components/block-drop-zone/index.js

```diff
@@ -84,51 +79,55 @@ class BlockDropZone extends Component {
 	}
 
 	render() {
-		const { isLocked } = this.props;
+		const { isLocked, index } = this.props;
 		if ( isLocked ) {
 			return null;
 		}
+		const isAppender = index === undefined;
 
 		return (
 			<DropZone
 				onFilesDrop={ this.onFilesDrop }
 				onHTMLDrop={ this.onHTMLDrop }
-				onDrop={ this.onDrop }
+				onDrop={ isAppender ? undefined : this.onDrop }
 			/>
 		);
 	}
 }
 
 export default compose(
-	connect(
-		undefined,
-		( dispatch, ownProps ) => {
-			return {
-				insertBlocks( blocks, insertIndex ) {
-					const { rootUID, layout } = ownProps;
-
-					if ( layout ) {
-						// A block's transform function may return a single
-						// transformed block or an array of blocks, so ensure
-						// to first coerce to an array before mapping to inject
-						// the layout attribute.
-						blocks = castArray( blocks ).map( ( block ) => (
-							cloneBlock( block, { layout } )
-						) );
-					}
-
-					dispatch( insertBlocks( blocks, insertIndex, rootUID ) );
-				},
-				updateBlockAttributes( ...args ) {
-					dispatch( updateBlockAttributes( ...args ) );
-				},
-				moveBlockToPosition( uid, fromRootUID, index ) {
-					const { rootUID, layout } = ownProps;
-					dispatch( moveBlockToPosition( uid, fromRootUID, rootUID, layout, index ) );
-				},
-			};
-		},
-	),
+	withDispatch( ( dispatch, ownProps ) => {
```

Comment:
Since I touched the file, I refactored it to use `withDispatch` instead of `connect`